File: CPP/BOJ-S5-4659-Pronounceable_password.cpp
#include <bits/stdc++.h>
using namespace std;

string s;
int ccnt, vcnt;

bool isVowel(int idx)
{
  return (idx == 'a' || idx == 'e' || idx == 'i' || idx == 'o' || idx == 'u');
}

int main()
{
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  cout.tie(NULL);
  // freopen("input.txt", "r", stdin);
  // freopen("output.txt", "w", stdout);

  while (true)
  {
    cin >> s;
    if (s == "end")
      break;

    ccnt = vcnt = 0;
    bool flag = 0;
    bool is_include_v = 0;
    int prev = -1;

    for (int i = 0; i < s.size(); ++i)
    {
      int idx = s[i];
      if (isVowel(idx))
        ccnt = 0, ++vcnt, is_include_v = 1;
      else
        vcnt = 0, ++ccnt;

      if (vcnt == 3 || ccnt == 3)
        flag = 1;

      if (i >= 1 && (prev == idx) && (idx != 'e' && idx != 'o'))
        flag = 1;

      prev = idx;
    }

    if (is_include_v == 0)
      flag = 1;

    if (flag)
      cout << "<" << s << ">" << " is not acceptable.\n";
    else
      cout << "<" << s << ">" << " is acceptable.\n";
  }

  return 0;
}

File: JS/BOJ-S5-4659-Pronounceable_password.js
'use strict';

const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : '../input.txt';
let input = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split('\n')
  .map((v) => v.trimEnd());

let output = [];
const solution = (input) => {
  const isVowel = (s) => {
    return s === 'a' || s === 'o' || s === 'e' || s === 'u' || s === 'i';
  };

  const outputStr = (b, s) => {
    output.push(`<${s}> is${b ? '' : ' not'} acceptable.`);
  };

  input.forEach((v) => {
    if (v === 'end') return;

    let vcnt = 0,
      ccnt = 0,
      isIncludeV = false;
    let prev = '';

    for (let i = 0; i < v.length; ++i) {
      if (isVowel(v[i])) {
        ++vcnt, (ccnt = 0);
        isIncludeV = true;
      } else {
        (vcnt = 0), ++ccnt;
      }

      if (prev === v[i] && v[i] !== 'o' && v[i] !== 'e') {
        outputStr(false, v);
        return;
      }

      if (vcnt >= 3 || ccnt >= 3) {
        outputStr(false, v);
        return;
      }

      prev = v[i];
    }

    if (!isIncludeV) {
      outputStr(false, v);
      return;
    }

    outputStr(true, v);
  });
};

solution(input);
fs.writeFileSync('/dev/stdout', output.join('\n').trimEnd());
// fs.writeFileSync(
//   '../output.txt',
//   output.join('\n').trimEnd(),
//   (err) => console.error
// );
